src/lsm/scan_lookup.zig

```diff
@@ -32,28 +32,12 @@ pub fn ScanLookupType(
         pub const Callback = *const fn (*ScanLookup, []const Object) void;
 
         const LookupWorker = struct {
+            index: u8,
             scan_lookup: *ScanLookup,
             lookup_context: Groove.ObjectTree.LookupContext = undefined,
             index_produced: ?usize = null,
         };
 
-        /// Since the workload is always sorted by timestamp,
-        /// adjacent objects are often going to be in the same table-value block.
-        /// The grid is aware when N lookups ask for the same grid block concurrently,
-        /// and queues up the reads internally such that they actually hit the storage once.
-        ///
-        /// To maximize IO utilization, we allow at least `Grid.read_iops_max` lookups to run,
-        /// up to an arbitrary constant based on the maximum number of objects per block.
-        /// Reasoning: the larger the block size is, higher is the probability of multiple
-        /// lookups hitting the same grid block:
-        const lookup_workers_max = @max(
-            stdx.div_ceil(
-                @divFloor(constants.block_size, @sizeOf(Object)),
-                Grid.read_iops_max,
-            ),
-            Grid.read_iops_max,
-        );
-
```

Comment:
This logic was removed:
- It simplified the mental model of always handling the max of `Grid.read_iops_max` parallel lookups.
- In practice, the benchmark didn't show any performance improvement.
- Reducing the number of `[max]LookupWorker` in the array also **reduced the size of the binary** :exploding_head: